cfme/tests/cloud_infra_common/test_tag_objects.py

```diff
@@ -5,58 +5,48 @@
 from cfme.cloud.keypairs import KeyPair
 from cfme.cloud.provider import CloudProvider
 from cfme.infrastructure.provider import InfraProvider
-from cfme.fixtures.provider import setup_one_or_skip
-from cfme.utils.providers import ProviderFilter
+from cfme.markers.env_markers.provider import ONE_PER_CATEGORY
 from cfme.utils.appliance.implementations.ui import navigate_to
 
 pytestmark = [
-    pytest.mark.tier(2)
+    pytest.mark.tier(2),
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([InfraProvider, CloudProvider], scope='module', selector=ONE_PER_CATEGORY)
 ]
 
 
 # tuples of (collection_name, destination)
 # get_collection_entity below processes these to navigate and find an entity
 # so use (collection_name, None, None) if the thing being tested uses BaseEntity/BaseCollection
 # Once everything is converted these should be flattened to just collection names list
+
 infra_test_items = [
     ('infra_provider', None),  # no param_class needed, provider returned directly
-    ('infra_vms', 'ProviderVms'),
-    ('infra_templates', 'ProviderTemplates'),
+    ('infra_vms', True),
+    ('infra_templates', True),
     ('hosts', None),
     ('clusters', None),
     ('datastores', None)
 ]
-
 cloud_test_items = [
     ('cloud_provider', None),  # no param_class needed, provider returned directly
-    ('cloud_instances', 'Instances'),
+    ('cloud_instances', True),
     ('cloud_flavors', None),
     ('cloud_av_zones', None),
     ('cloud_tenants', None),
     ('cloud_keypairs', None),
-    ('cloud_images', 'Images')
+    ('cloud_images', True)
 ]
 
 
-@pytest.fixture(scope='module')
-def infra_provider(request):
-    prov_filter = ProviderFilter(classes=[InfraProvider])
-    return setup_one_or_skip(request, filters=[prov_filter])
-
-
-@pytest.fixture(scope='module')
-def cloud_provider(request):
-    prov_filter = ProviderFilter(classes=[CloudProvider],
-                                 required_fields=[['provisioning', 'stacks']])
-    return setup_one_or_skip(request, filters=[prov_filter])
-
-
 def get_collection_entity(appliance, collection_name, destination, provider):
     if collection_name in ['infra_provider', 'cloud_provider']:
         return provider
     else:
         collection = getattr(appliance.collections, collection_name)
-        view = navigate_to(provider, destination) if destination else navigate_to(collection, 'All')
+        collection.filters = {'provider': provider}
```

Comment:
I think since this test collection and parametrization has changed enough that its worth mixing it up a bit.

I think it would be a lot more clear to define the destination string within the test_items tuples instead of inferring them here.

Reply:
done